### DISCUSSION TITLE
Should media info (album art and track / album) be seen in players exposed to Home Assistant?

### DISCUSSION THREAD

**Discussion by sockmonkey0223** [music-assistant/support#2709] [Q&A] [answered]
I've just set up MASS and working on getting things configured.

In the documentation, it is recommended to use (in HASS) the player entities exposed by the MASS integration because they are optimized for music. However, I've found that those entities don't show the album art, track, or album info when playing. The native HASS entities for the same players do show this information.

Is this expected behavior or have I misconfigured something?

**Comment by chasut** [comment]
It does (mostly) work for me.  I am interested to see more discussions/information along the lines of best practices and different ways to optimize the experience on the HA side.

It might be helpful to provide some additional information on the Providers and Players in use, as the behavior may be slightly different depending on the specific circumstances.

My Providers are Apple Music, Tune-in Radio, and the HA Radio Browser.  My players are all airplay.  And I do have instances where they are duplicated on the HA side through the Apple TV integration.  It is as you describe that these "duplicate" entities do show the correct meta-data for the playing track.

What I have done is to create an airplay group player (I call it Home Audio) in MA.  That group player includes all of the other "players" that I want to control in my whole-home audio setup.  When I select where I want to play the audio, it then also shows the correct album and track information on the Home Audio entity.  So all controls and my custom UI are centered around that media_player entity in HA.

Using the MA services and the MA entities in HA, I have been able to create a simple but effective UI for whole home audio that has a reasonably high WAF.  My only issue is that on odd occasions  the group feature when selecting players gets sideways(when de-selecting the first player selected).  Everything else works great.

That said, I am not certain I am doing it right.  But it is mostly good.  It is still early so I expect more to experiment and come up with how to make this work.  Your post looks as good as any to start that discussion.

**Reply by OzGav** [reply]
@sockmonkey0223 what player providers are you using?

**Reply by sockmonkey0223** [reply]
Players: AirPlay & Sonos.
Music Providers: Filesystem and Apple Music.

Specifically when seeing this "issue" (no media info in HA) I tested on:
1) A Sonos device using AirPlay
2) A Sonos device using Sonos
3) An AirPort Express device using AirPlay.

I've had to disable the Apple Music provider because it seemed to keep crashing _HA_ but I haven't had time to dig into it.

**Reply by OzGav** [reply]
The Sonos provider is currently being rewritten. Not sure about the Airport Express. Is it one of the first ones? If so it is pretty old...

**Reply by OzGav** [reply, marked as answer]
The new Sonos provider is in the beta channel. You can try that and see if this problem is resolved.
